File: python/copass-management/src/copass_management/tools/start_integration_connect.py
from __future__ import annotations

from typing import Any, Dict, TYPE_CHECKING

if TYPE_CHECKING:
    from copass_management.registrar import ToolContext


async def start_integration_connect(
    ctx: "ToolContext", input: Dict[str, Any],
) -> Any:
    app_slug = str(input["app_slug"])
    kwargs: Dict[str, Any] = {
        "success_redirect_uri": str(input.get("success_redirect_uri") or ""),
        "error_redirect_uri": str(input.get("error_redirect_uri") or ""),
    }
    if input.get("webhook_uri") is not None:
        kwargs["webhook_uri"] = str(input["webhook_uri"])
    return await ctx.client.integrations.connect(
        ctx.sandbox_id, app_slug, **kwargs,
    )

File: typescript/packages/management/src/tools/start_integration_connect.ts
import type { ConnectRequest } from '@copass/core';

import type { ToolContext, ToolHandler } from '../registrar.js';

export const startIntegrationConnect: ToolHandler = async (
  ctx: ToolContext,
  input: Record<string, unknown>,
) => {
  const appSlug = String(input.app_slug);
  const successRedirect =
    typeof input.success_redirect_uri === 'string'
      ? input.success_redirect_uri
      : '';
  const errorRedirect =
    typeof input.error_redirect_uri === 'string'
      ? input.error_redirect_uri
      : '';
  const request: ConnectRequest & { webhookUri?: string } = {
    success_redirect_uri: successRedirect,
    error_redirect_uri: errorRedirect,
  };
  if (typeof input.webhook_uri === 'string') {
    request.webhookUri = input.webhook_uri;
  }
  return ctx.client.integrations.connect(ctx.sandboxId, appSlug, request);
};
